e2e/test/cli_test.go

```diff
@@ -0,0 +1,138 @@
+package e2e_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/omni-network/omni/cli/cmd"
+	"github.com/omni-network/omni/lib/anvil"
+	"github.com/omni-network/omni/lib/cchain/provider"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+
+	"github.com/cometbft/cometbft/rpc/client/http"
+
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/stretchr/testify/require"
+)
+
+// exeCLI will execute provided command with the arguments and return an error in case
+// execution fails, or command output as string in case of success.
+func exeCLI(args ...string) (string, error) {
+	outBuf := new(bytes.Buffer)
+	errBuf := new(bytes.Buffer)
+
+	root := cmd.New()
+	root.SetOut(outBuf)
+	root.SetErr(errBuf)
+
+	root.SetArgs(args)
+	if err := root.Execute(); err != nil {
+		return "", err
+	}
+	if errBuf.Len() > 0 {
+		return "", errors.New(errBuf.String())
+	}
+
+	return outBuf.String(), nil
+}
+
+// TestValidatorCommands tests multiple CLI operator commands in sequence.
+// The test runs the following commands:
+// - operator create-validator creates a new validator and makes sure the validator is added to the consensus chain
+// - operator delegate increases the newly created validator stake and makes sure its power is increased
+//
+// Since they rely first on validator being created it must be run as a unit.
+func TestValidatorCommands(t *testing.T) {
+	t.Parallel()
+
+	testNetwork(t, func(t *testing.T, network netconf.Network, endpoints xchain.RPCEndpoints) {
```

Comment:
why not have a nice `testCLI` with signature: `func(t *testing.T, omniEVMCl ethclient.Client, rpcClient *comethttp.HTTP)`?

You can do queries and stuff to both omni evm and omni conensus using those clients, construct xproviders or cproviders if required, and they both expose internal addresses that you should pass to the cli command.

Reply:
we can, I was trying to wrap things up and we can do that when we test more commands and know what will be used by other commands as well, I can change this here if you think it's better